executor/src/witgen/machines/mod.rs

```diff
@@ -54,24 +54,26 @@ pub trait Machine<'a, T: FieldElement>: Send + Sync {
         );
     }
 
-    /// Returns Some(..) if this machine can alway fully process a call via the given
+    /// Returns (true, _) if this machine can alway fully process a call via the given
     /// identity, the set of known arguments and a list of range constraints
     /// on the parameters. Note that the range constraints can be imposed both
     /// on inputs and on outputs.
-    /// If this returns Some(..), then corresponding calls to `process_lookup_direct`
+    /// If this returns (true, _), then corresponding calls to `process_lookup_direct`
     /// are safe.
-    /// The value returned inside the option is a vector of range constraints on the arguments,
+    /// The second return value is a vector of range constraints on the arguments,
     /// which again can be imposed both on inputs and on outputs.
+    /// The returned range constraints can be used by the caller even if the first
+    /// return value is false.
     /// The function requires `&mut self` because it usually builds an index structure
     /// or something similar.
     fn can_process_call_fully(
         &mut self,
         _can_process: impl CanProcessCall<T>,
         _identity_id: u64,
         _known_arguments: &BitVec,
-        _range_constraints: &[RangeConstraint<T>],
-    ) -> Option<Vec<RangeConstraint<T>>> {
-        None
+        range_constraints: &[RangeConstraint<T>],
+    ) -> (bool, Vec<RangeConstraint<T>>) {
+        (false, range_constraints.to_vec())
```

Comment:
it looks a bit weird that every place where (false, constraints) is returned, the constraints argument needs to be `to_vec()` and returned... or is there some case where something different might be returned? Otherwise, maybe this should be handled at the calling site?

Reply:
Yes, the fixed lookup actually returns different constraints.

Reply:
Here: https://github.com/powdr-labs/powdr/pull/2444/files#diff-20b382dd5fa786b248177294f7c7497094f1ce43dbc78af3a2328024a6066e93R379

Reply:
hmm i see... maybe `can_process_call_fully` could take a `Vec` directly and just return it then

Reply:
good idea!